deploy/example-deployment.yaml

```diff
@@ -1,34 +1,100 @@
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: custom-metrics
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: custom-metrics-apiserver
+  namespace: custom-metrics
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: custom-metrics:system:auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: custom-metrics-apiserver
+  namespace: custom-metrics
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: RoleBinding
+metadata:
+  name: custom-metrics-auth-reader
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
+subjects:
+- kind: ServiceAccount
+  name: custom-metrics-apiserver
+  namespace: custom-metrics
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: custom-metrics-resource-reader
+rules:
+- apiGroups:
+  - ""
+  resources:
```

Comment:
we should have a note that the default policy restricts the resources to this set.  From a theoretical perspective, this could be `*`.  From a security perspective, that's probably sub-optimal.  What we really want is to be able to list the names of things w/o getting the actual contents.  This allows us to work with any resource, while limiting the attack surface in case of compromise.  It wouldn't be perfect (names can still leak important details), but it's *much* better than list itself.  @deads2k @liggitt either of you have opinions on this matter?

@luxas: also, I don't think we need the `get` permission, IIRC, just `list`.